components/purchase_modal/purchase_modal.tsx

```diff
@@ -290,6 +326,72 @@ export default class PurchaseModal extends React.PureComponent<Props, State> {
         });
     }
 
+    paymentFooterText = () => {
+        const normalPaymentText = (
+            <FormattedMessage
+                defaultMessage={'Payment begins: {beginDate}'}
+                id={'admin.billing.subscription.paymentBegins'}
+                values={{
+                    beginDate: getNextBillingDate(),
+                }}
+            />
+        );
+
+        let payment = normalPaymentText;
+        if ((!this.props.isFreeTrial && this.state.currentProduct?.name.toLowerCase().includes('starter')) &&
+                !this.state.selectedProduct?.name.toLowerCase().includes('starter')) {
```

Comment:
We can generalize this condition to be: `if currentProduct.billing_scheme == flat_rate && selectedProduct,billing_scheme== per_user`

So in general, any time we are going from a flat-fee to user-based there will be prorated charges.

Reply:
ok